### PR TITLE
Test null, empty and malformed data corner cases

### PR DESCRIPTION
**Pull request by onufryk**

### CODE REVIEW

File: core-api/src/test/java/com/optimizely/ab/OptimizelyBuilderTest.java [hunk 1]
```diff
@@ -180,6 +180,7 @@ public void builderThrowsConfigParseExceptionForEmptyDatafile() throws Exception
     @Test
     public void builderThrowsConfigParseExceptionForInvalidDatafile() throws Exception {
         thrown.expect(ConfigParseException.class);
-        Optimizely.builder("{invalidDatafile}", mockEventHandler).build();
+        String invalidJson = "{invalidDatafile}";
```
Comment (vraja2) [inline core-api/src/test/java/com/optimizely/ab/OptimizelyBuilderTest.java]: since this was just for rebuilding, can we change this back?
